### DISCUSSION TITLE
Duplicate name, rejecting enrollment. Help

### DISCUSSION THREAD

**Discussion by hexclann** [wazuh/wazuh#13571] [Q&A] [answered]
Hello, I'm running Wazuh on Ubuntu, I'm getting an error `Duplicate name, rejecting enrollment...` when I try to install the agent twice. I have set the configuration as indicated in https://github.com/wazuh/wazuh/issues/3674. But no luck
My setup information:

#### Wazuh-manager:
Ubuntu 20.04 LTS
16GB RAM
AMD64 architecture
Running in Digitalocean droplet

#### Wazuh-agent:
Ubuntu 20.04 LTS
1GB RAM
AMD64 architecture
Running in Digitalocean droplet

#### Manager logs:
```
...
2022/05/24 14:54:02 wazuh-authd: INFO: New connection from 64.x.x.x
2022/05/24 14:54:02 wazuh-authd: INFO: Received request for a new agent (ubuntu-s-1vcpu-1gb-intel-blr1-01) from: 64.x.x.x
2022/05/24 14:54:02 wazuh-authd: WARNING: Duplicate name 'ubuntu-s-1vcpu-1gb-intel-blr1-01', rejecting enrollment. Agent '002' key already exists on the manager.
...
```

#### Agent logs:

```
...
2022/05/24 09:48:13 wazuh-agentd: INFO: Requesting a key from server: test.example.com
2022/05/24 09:48:13 wazuh-agentd: INFO: No authentication password provided
2022/05/24 09:48:13 wazuh-agentd: INFO: Using agent name as: ubuntu-s-1vcpu-1gb-intel-blr1-01
2022/05/24 09:48:13 wazuh-agentd: INFO: Waiting for server reply
2022/05/24 09:48:13 wazuh-agentd: ERROR: Duplicate agent name: ubuntu-s-1vcpu-1gb-intel-blr1-01 (from manager)
2022/05/24 09:48:13 wazuh-agentd: ERROR: Unable to add agent (from manager)
...
```

#### Manager auth config

```
<auth>
    <disabled>no</disabled>
    <port>40272</port>
    <use_source_ip>no</use_source_ip>
    <force_insert>yes</force_insert>
    <force_time>0</force_time>
    <purge>yes</purge>
    <use_password>no</use_password>
    <ciphers>HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH</ciphers>
    <!-- <ssl_agent_ca></ssl_agent_ca> -->
    <ssl_verify_host>no</ssl_verify_host>
    <ssl_manager_cert>etc/sslmanager.cert</ssl_manager_cert>
    <ssl_manager_key>etc/sslmanager.key</ssl_manager_key>
    <ssl_auto_negotiate>no</ssl_auto_negotiate>
 </auth>
```

The ports are changed and updated in the agent config. I can see the new agents in the dashboard but unable to register an existing agent. How can I make the manager register an already existing agent by discarding the old key?

**Comment by xrisbarney** [comment]
Hi @hexclann,

Thank you for using Wazuh!

- To remove an agent, please follow the guide [here](https://documentation.wazuh.com/current/user-manual/agents/remove-agents/remove.html)
- To enroll an agent with a duplicate name, you can specify the agent_name variable in the enrollment section of the agent ossec.conf file. The documentation for agent enrollment is [here](https://documentation.wazuh.com/current/user-manual/agent-enrollment/via-agent-configuration/windows-endpoint.html).

Please let me know if this helps. Best.

**Comment by Selutario** [comment, marked as answer]
Fixed in:
- [Wazuh v4.3.3](https://github.com/wazuh/wazuh/releases/tag/v4.3.3). [Release notes](https://documentation.wazuh.com/current/release-notes/release-4-3-3.html).
